src/test/java/com/epam/tc/hw3/page/IndexPage.java

```diff
@@ -0,0 +1,76 @@
+package com.epam.tc.hw3.page;
+
+import java.util.List;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class IndexPage {
+
+    private static final int WAIT_TIMEOUT_SEC = 5;
+
+    private HeaderNavBar headerNavBar;
+    private LeftNavBar leftNavBar;
+
+    @FindBy(css = "li.uui-profile-menu")
+    private WebElement loginDropdown;
+
+    @FindBy(id = "name")
+    private WebElement searchInputName;
+
+    @FindBy(id = "password")
+    private WebElement searchInputPassword;
+
+    @FindBy(id = "login-button")
+    private WebElement loginButton;
+
+    @FindBy(css = "span[id='user-name'].hidden")
+    private List<WebElement> notLoggedInUserNames;
+
+    @FindBy(id = "user-name")
+    private WebElement loggedInUserName;
+
+    public IndexPage(WebDriver driver) {
+
+        PageFactory.initElements(driver, this);
+
+        headerNavBar = new HeaderNavBar();
+        PageFactory.initElements(driver, headerNavBar);
+
+        leftNavBar = new LeftNavBar();
+        PageFactory.initElements(driver, leftNavBar);
+    }
+
+    public IndexPage loginUser(WebDriver driver, String userName, String userPassword) {
+        if (notLoggedInUserNames.size() != 0) {
+            new WebDriverWait(driver, WAIT_TIMEOUT_SEC)
```

Comment:
It is better to create a dedicated class which would be responsible for WebDriverWait

Reply:
added WebDriverWaitClass with respective methods